### PR TITLE
AT-90 Changed response to include status and updated tests accordingly

### PR DESCRIPTION
**Pull request by grantwhiteman**
Changed response to include status i.e. 'success'/'failure'/'pending'

https://dfacademy.atlassian.net/jira/software/projects/AT/boards/4?issueParent=10138&selectedIssue=AT-90

### CODE REVIEW

File: cypress/integration/FileUpload/ModuleChallenge.spec.js [hunk 1]
```diff
@@ -70,7 +72,9 @@ describe("Module Challenge upload feature tests : ", function () {
           const enc = new TextDecoder("utf-8");
           const text = enc.decode(res.body);
           const parsedData = JSON.parse(text);
-          expect(parsedData.response[0]).to.eq("Headers: [challengeName,language,studentScore,coachScore,dueDate] does not match any valid headers");
+          console.log(parsedData);
```
Comment (dearshrewdwit) [inline cypress/integration/FileUpload/ModuleChallenge.spec.js:75]: remove debugging statement

File: src/fileUpload/fileUpload.js [hunk 2]
```diff
@@ -65,11 +69,14 @@ class FileUploader {
   }   
 
   async addToDatabase () {
-    if (this.errors.length === 0) {
+    if (this.result.errors.length === 0) {
       await this.table.model.bulkCreate(this.data);
-      return ["Updated the database successfully."];
+      this.result.status = "success";
+      this.result.errors = ["Updated the database successfully."];
```
Comment (dearshrewdwit) [inline src/fileUpload/fileUpload.js:75]: this is no longer needed if you store a status property.

(On the front end you can map the status to anything you want)

File: src/fileUpload/fileUpload.js [hunk 3]
```diff
@@ -65,11 +69,14 @@ class FileUploader {
...
+      return this.result;
     } else {
-      return this.errors;
+      this.result.status = "failure";
+      return this.result;
```
Comment (dearshrewdwit) [inline src/fileUpload/fileUpload.js:79]: Given you're returning it regardless, move the return outside the if statement

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
Better. See comments.

FYI this looks like it might compete with other work in your team - upload history and some refactors

**Review by benclayton98 — DISMISSED** [review]
LGTM

File: src/fileUpload/fileUpload.js [hunk 4]
```diff
@@ -69,23 +70,24 @@ class FileUploader {
   }
 
   async addToDatabase () {
-    if (this.errors.length === 0) {
+    if (this.result.errors.length === 0) {
       await this.table.model.bulkCreate(this.data);
-      this.status = "Success";
+      this.result.status = "success";
+      console.log(`erros at addtodatabase for success: ${this.result.errors}`);
```
Comment (dearshrewdwit) [inline src/fileUpload/fileUpload.js:76]: 🤨

File: src/fileUpload/fileUpload.js [hunk 5]
```diff
@@ -69,23 +70,24 @@ class FileUploader {
...
       await this.setHistory();
-      return ["Updated the database successfully."];
     } else {
-      this.status = "Failure";
+      this.result.status = "failure";
+      console.log(`erros at addtodatabase for failure: ${this.result.errors}`);
```
Comment (dearshrewdwit) [inline src/fileUpload/fileUpload.js:80]: 🦅

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

**Review by dearshrewdwit — APPROVED** [review]

**Review by benclayton98 — APPROVED** [review]
LBTM

**Pull request merged**